### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by imaginetheheadline** [marcpllew/tic-tac-toe#1]
# Project 1

Nice work, Marc! Once again, I like all the planning you did. I know you had a bit of help here, so I don't have too many comments. It looks good.

## HTML

HTML looks very clean. Nice use of data attributes!

## CSS

Nice and simple, I like it!

## JavaScript

JavaScript also looks good and to the point!

### Functions

Nice use of functions:

```js
function drawMessage(message) {
    statusDisplay.innerText = message;
}
```

Makes it easier for your self.

### Winning Message

This is really great implementation of who wins. Since O always starts, Player X wins on an even turn. Brilliant :) Same thing for the `displayCurrentPlayer`.

### Board Array

Nice work on having the board in a 2D array in JS. This means you can easily access the values there without needing the DOM :)

```js
 for (i = 0; i < boxes.length; i++) {
        // looping through boxes
        // console.log(boxes[i].dataset.row);
        board[boxes[i].dataset.row][boxes[i].dataset.column] =
            boxes[i].textContent;
        // will return all the contents of 'boxes'. dataset is accessing the content of row and column, and populating the board array
    }
```

Ah I see, you're using it to get data from the DOM after each click. Does this all make sense?

### Preventing clicks on same box

Nice work.

### Winning Conditions

Looks like you've covered them. How was the 2D array?

### Restart Game

Also looks good, just running the same code as the setup and resetting the content of each box. :)

**Comment by marcpllew** [comment]
Yes the code block you highlighted and asked if it makes sense was the hardest bit of code in the JS for me to get my head around. I sat infront of this a few times going through it to try and get my head around it properly. Even now i have to really think hard again to under stand it. Probably could use a different method that is perhaps a little longer but would be easier to understand.
I liked the use of the 2D array as a virtual board. It made things easy to understand as it was quite visual.

**Comment by imaginetheheadline** [comment]
Yeah, gotcha. You're basically giving each box row and column coordinates and then grabbing those to access the 2D array at the corresponding location. Nice to hear it helped to visualise it :)
